packages/http-client-python/emitter/src/emitter.ts

```diff
@@ -178,39 +194,27 @@ export async function $onEmit(context: EmitContext<PythonEmitterOptions>) {
             black.BlackScriptPlugin(output_folder=outputFolder, **commandArgs).process()
       
           await main()`;
-        await pyodide.runPythonAsync(pythonCode, { globals });
+      await pyodide.runPythonAsync(pythonCode, { globals });
+    } else {
+      // here we run with native python
+      let venvPath = path.join(root, "venv");
+      if (fs.existsSync(path.join(venvPath, "bin"))) {
+        venvPath = path.join(venvPath, "bin", "python");
+      } else if (fs.existsSync(path.join(venvPath, "Scripts"))) {
+        venvPath = path.join(venvPath, "Scripts", "python.exe");
       } else {
-        // here we run with native python
-        let venvPath = path.join(root, "venv");
-        if (fs.existsSync(path.join(venvPath, "bin"))) {
-          venvPath = path.join(venvPath, "bin", "python");
-        } else if (fs.existsSync(path.join(venvPath, "Scripts"))) {
-          venvPath = path.join(venvPath, "Scripts", "python.exe");
-        } else {
-          reportDiagnostic(program, {
-            code: "pyodide-flag-conflict",
-            target: NoTarget,
-          });
-        }
-        commandArgs["output-folder"] = outputDir;
-        commandArgs["cadl-file"] = yamlPath;
-        const commandFlags = Object.entries(commandArgs)
-          .map(([key, value]) => `--${key}=${value}`)
-          .join(" ");
-        const command = `${venvPath} ${root}/eng/scripts/setup/run_tsp.py ${commandFlags}`;
-        execSync(command, { stdio: [process.stdin, process.stdout] });
+        reportDiagnostic(program, {
+          code: "pyodide-flag-conflict",
+          target: NoTarget,
+        });
       }
-    } catch (error: any) {
-      const errStackStart =
-        "========================================= error stack start ================================================";
-      const errStackEnd =
-        "========================================= error stack end ================================================";
-      const errStack = error.stack ? `\n${errStackStart}\n${error.stack}\n${errStackEnd}` : "";
-      reportDiagnostic(program, {
-        code: "unknown-error",
-        target: NoTarget,
-        format: { stack: errStack },
-      });
+      commandArgs["output-folder"] = outputDir;
```

Comment:
why are we doing this change in this pr? I'm overall fine with it, just wondering

Reply:
For known warning/error, we use reportDiagnostics so that we could report as many error/warningas possible one time. For unexpected error, there is ask that emitter shall not crash so I add `try/catch` in the outermost.